src/main/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.java

```diff
@@ -86,10 +84,46 @@ public static boolean containsAnnotation(final DetailAST ast) {
         return holder != null && holder.findFirstToken(TokenTypes.ANNOTATION) != null;
     }
 
+    /**
+     * Checks if the given AST element is annotated with any of the specified annotations.
+     *
+     * <p>
+     * This method accepts both simple and fully-qualified names,
+     * e.g. "Override" will match both java.lang.Override and Override.
+     * </p>
+     *
+     * @param ast The type or method definition.
+     * @param annotations A collection of annotations to look for.
+     * @return {@code true} if the given AST element is annotated with
+     *                      at least one of the specified annotations;
+     *                      {@code false} otherwise.
+     */
+    public static boolean containsAnnotation(DetailAST ast, List<String> annotations) {
+        if (ast == null) {
+            throw new IllegalArgumentException(THE_AST_IS_NULL);
+        }
+
+        boolean result = false;
+
+        if (annotations != null && !annotations.isEmpty()) {
+            result = getAnnotation(ast, annotationNode -> {
+                DetailAST identNode = annotationNode.findFirstToken(TokenTypes.IDENT);
+                if (identNode == null) {
+                    identNode = annotationNode.findFirstToken(TokenTypes.DOT)
+                            .findFirstToken(TokenTypes.IDENT);
+                }
+
+                return annotations.contains(identNode.getText());
+            }) != null;
```

Comment:
this is impossible to read properly.
please store result of `getAnnotation` in some variable to reuse in "result = annotation != null"